StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/ElementOrderCodeFixProvider.cs

```diff
@@ -0,0 +1,196 @@
+﻿namespace StyleCop.Analyzers.OrderingRules
+{
+    using System.Collections.Immutable;
+    using System.Composition;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.CodeActions;
+    using Microsoft.CodeAnalysis.CodeFixes;
+    using Microsoft.CodeAnalysis.CSharp;
+    using Microsoft.CodeAnalysis.CSharp.Syntax;
+    using StyleCop.Analyzers.Helpers;
+
+    /// <summary>
+    /// Implements code fixes for element ordering rules.
+    /// </summary>
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(ElementOrderCodeFixProvider))]
+    [Shared]
+    public class ElementOrderCodeFixProvider : CodeFixProvider
+    {
+        private static readonly ImmutableArray<string> FixableDiagnostics =
+            ImmutableArray.Create(
+                SA1201ElementsMustAppearInTheCorrectOrder.DiagnosticId,
+                SA1202ElementsMustBeOrderedByAccess.DiagnosticId,
+                SA1203ConstantsMustAppearBeforeFields.DiagnosticId,
+                SA1204StaticElementsMustAppearBeforeInstanceElements.DiagnosticId,
+                SA1214StaticReadonlyElementsMustAppearBeforeStaticNonReadonlyElements.DiagnosticId,
+                SA1215InstanceReadonlyElementsMustAppearBeforeInstanceNonReadonlyElements.DiagnosticId);
+
+        private static bool checkElementType;
+        private static bool checkAccessLevel;
+        private static bool checkConst;
+        private static bool checkStatic;
+        private static bool checkReadonly;
+
+        /// <inheritdoc/>
+        public override ImmutableArray<string> FixableDiagnosticIds => FixableDiagnostics;
+
+        /// <inheritdoc/>
+        public override FixAllProvider GetFixAllProvider()
+        {
+            return FixAll.Instance;
+        }
+
+        /// <inheritdoc/>
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            SemanticModel semanticModel;
+            context.Document.TryGetSemanticModel(out semanticModel);
+            if (semanticModel != null)
+            {
+                checkElementType = !semanticModel.Compilation.IsAnalyzerSuppressed(SA1201ElementsMustAppearInTheCorrectOrder.DiagnosticId);
```

Comment:
I would extract this code to allow for reuse between these lines and lines 170 and above